clients/src/main/java/org/apache/kafka/common/requests/ElectPreferredLeadersRequest.java

```diff
@@ -0,0 +1,160 @@
+
+    private static final String TIMEOUT_KEY_NAME = "timeout";
+    private static final String TOPIC_PARTITIONS_KEY_NAME = "topic_partitions";
+    private static final String PARTITIONS_KEY_NAME = "partitions";
+
+    public static final Schema ELECT_PREFERRED_LEADERS_REQUEST_V0 = new Schema(
+            new Field(TOPIC_PARTITIONS_KEY_NAME, ArrayOf.nullable(
+                    new Schema(
+                            TOPIC_NAME,
+                            new Field(PARTITIONS_KEY_NAME,
+                                    new ArrayOf(INT32),
+                                    "The partitions of this topic whose preferred leader should be elected")))),
+            new Field(TIMEOUT_KEY_NAME, INT32, "The time in ms to wait for the elections to be completed.")
+    );
+
+    public static Schema[] schemaVersions() {
+        return new Schema[]{ELECT_PREFERRED_LEADERS_REQUEST_V0};
+    }
+
+    public static class Builder extends AbstractRequest.Builder {
+
+        private final Set<TopicPartition> topicPartitions;
+        private final int timeout;
+
+        public Builder(Collection<TopicPartition> topicPartitions, int timeout) {
+            super(ApiKeys.ELECT_PREFERRED_LEADERS);
+            this.topicPartitions = topicPartitions != null ? new HashSet<>(topicPartitions) : (Set) null;
+            this.timeout = timeout;
+        }
+
+        @Override
+        public ElectPreferredLeadersRequest build(short version) {
+            return new ElectPreferredLeadersRequest(version, topicPartitions, timeout);
+        }
+    }
+
+    private final Set<TopicPartition> topicPartitions;
+    private final int timeout;
+
+    public ElectPreferredLeadersRequest(short version, Set<TopicPartition> topicPartitions, int timeout) {
+        super(version);
+        this.topicPartitions = topicPartitions;
+        this.timeout = timeout;
+    }
+
+    public ElectPreferredLeadersRequest(Struct struct, short version) {
+        super(version);
+        Object[] topicPartitionsArray = struct.getArray(TOPIC_PARTITIONS_KEY_NAME);
+        if (topicPartitionsArray != null) {
+            topicPartitions = new HashSet<>(topicPartitionsArray.length);
+            for (Object topicPartitionObj : topicPartitionsArray) {
+                Struct topicPartitionStruct = (Struct) topicPartitionObj;
+                String topicName = topicPartitionStruct.get(TOPIC_NAME);
+                Object[] partitionsArray = topicPartitionStruct.getArray(PARTITIONS_KEY_NAME);
+                if (partitionsArray != null) {
+                    for (Object partitionObj : partitionsArray) {
+                        Integer partition = (Integer) partitionObj;
+                        TopicPartition topicPartition = new TopicPartition(topicName, partition);
+                        topicPartitions.add(topicPartition);
+                    }
+                }
+            }
+        } else {
+            topicPartitions = null;
```

Comment:
Can we just use an empty list instead of null?  Currently it's not useful to pass an empty list anyway.